src/impl_ops.rs

```diff
@@ -117,13 +150,21 @@ where
     B: Clone,
     S: Data<Elem=A>,
     S2: Data<Elem=B>,
-    D: Dimension,
+    D: Dimension + BroadcastShape<E>,
     E: Dimension,
 {
-    type Output = Array<A, D>;
-    fn $mth(self, rhs: &'a ArrayBase<S2, E>) -> Array<A, D> {
-        // FIXME: Can we co-broadcast arrays here? And how?
-        self.to_owned().$mth(rhs)
```

Comment:
It needs to be mentioned that the number one thing I'd want to fix with this impl is not co-broadcasting, but removing the `.to_owned()` call - that is we don't want to traverse the lhs array twice (once to copy and once to do the arithmetic).